### PR TITLE
WIP: FS-121 Materiality Agent

### PR DESCRIPTION
**Pull request by IMladjenovic**
## Description
https://scottlogic.atlassian.net/browse/FS-121

## Changelog
* Materiality Agent
* Split Agent class into ChatAgent for Infer chat logic and Agent parent
  class for orchestrated Agent logic
* Convert Report Agent to new Agent style class
* Add materiality docs to github under library folder. Add libray folder
  to Dockerfile
* hard code company name to Astra Zeneca for now

### CODE REVIEW

File: backend/promptfoo/materiality_agent_config.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+description: "Test Dynamic Knowledge Graph Prompts"
```
Comment (mic-smith) [inline backend/promptfoo/materiality_agent_config.yaml:1]: Description needs updating

Comment (IMladjenovic) [reply backend/promptfoo/materiality_agent_config.yaml]: done

File: backend/src/llm/llm.py [hunk 2]
```diff
@@ -24,5 +45,22 @@ def get_instances(cls):
...
+        pass
+
+    @abstractmethod
+    def chat_with_file(
+        self,
+        model: str,
+        system_prompt: str,
+        user_prompt: str,
+        files_by_path: Optional[list[LLMFileFromPath]] = None,
+        files_by_stream: Optional[list[LLMFileFromBytes]] = None
```
[the remark this answers is not in the document]
Comment (dianaPrahoveanu-SL) [reply backend/src/llm/llm.py]: That sounds great to me!

**Review by dianaPrahoveanu-SL — APPROVED** [review]

File: backend/src/agents/materiality_agent.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+engine = PromptEngine()
+logger = logging.getLogger(__name__)
+
+
+class MaterialityAgent(Agent):
+    async def list_material_topics(self, company_name: str) -> dict[str, str]:
+        with open('./library/catalogue.json') as file:
+            catalogue = json.load(file)
+            files_json = await self.llm.chat(
```
Comment (mic-smith) [inline backend/src/agents/materiality_agent.py:17]: Nitpick: Worth unindenting from here to end of method, so we do the file read, close it at end of `with` and then do the llm calls etc?

**Review by mic-smith — APPROVED** [review]
Nice, looks good to me

**Pull request merged**